interbotix_ros_xseries/interbotix_xs_sdk/src/gripper_calib.cpp

```diff
@@ -0,0 +1,147 @@
+#include "interbotix_xs_sdk/xs_sdk_obj.h"
+#include <iostream>
+#include <fstream>
+/**
+ * @brief Class for gripper finger calibration.
+ */
+class Gripper_calibration{
+private:
+    ros::NodeHandle node;
+    ros::Publisher pub;
+    ros::ServiceClient client;
+    ros::Subscriber sub;
+    float prev_val = 0.0;
+    interbotix_xs_msgs::JointSingleCommand gripper_command_msg;
+    interbotix_xs_msgs::GripperCalib gripper_calib_srv;
+    std::string _gripper_name;
+public:
+    /**
+     * @brief Construct a new Gripper_calibration object.
+     * @param node_handle - Node object
+     * @param success  - Checker flag if node processes are succesful
+     * @param robot_name  - Name of the robot
+     * @param gripper_name  - Name of the gripper
+     */
+    Gripper_calibration(ros::NodeHandle* node_handle, bool &success, std::string robot_name, std::string gripper_name):node(*node_handle),_gripper_name(gripper_name)
+    {
+        pub = node.advertise<interbotix_xs_msgs::JointSingleCommand>("/"+robot_name+"/commands/joint_single",1000);
+        sub = node.subscribe("/"+robot_name+"/joint_states",1000,&Gripper_calibration::calib_callback,this);
+        client = node.serviceClient<interbotix_xs_msgs::GripperCalib>("/"+robot_name+"/gripper_calibration");
+        calib_complete_srv(0.0,_gripper_name);
+    }
+
+    /**
+     * @brief Returns the gripper index in joint states of the parsed gripper name
+     * @param names - list of joint names
+     * @return int  - index of the gripper name
+     */
+    int find_gripper_index(const std::vector<std::string>& names){
+        return std::find(names.begin(),names.end(),_gripper_name)-names.begin();
+    }
+
+    /**
+     * @brief Subscriber callback for performing gripper calibration
+     */
+    void calib_callback(const sensor_msgs::JointState &msg){
+        static int count = 0;
+        if(count%40==0){
+            int index = find_gripper_index(msg.name);
+            float curr_gripper_pos = msg.position[index];
+            ROS_INFO("I heard this position [%f]",msg.position[index]);
```

Comment:
Some of these ROS_INFO logging lines read like leftover debug messages for development. This one can probably be removed.

Reply:
Removed